packages/iTwinUI-react/src/core/Table/TableRowMemoized.tsx

```diff
@@ -50,22 +52,24 @@ export const TableRow = <T extends Record<string, unknown>>(props: {
   }, [isLast, onBottomReached, onRowInViewport, row.original]);
 
   const [rowRef, setRowRef] = React.useState<HTMLElement | null>();
-  const getIntersectionRoot = () => {
+  const intersectionRoot = React.useMemo(() => {
+    const rowParent = enableVirtualization
+      ? rowRef?.parentElement?.parentElement?.parentElement
+      : rowRef?.parentElement;
```

Comment:
Should we flip this to have a more dynamic check that recursively looks upwards for a scrollable container? `parentElement.parentElement.parentElement` looks hacky/fragile. Also `isTableBodyScrollable` sounds wrong now.

Reply:
so what you suggest instead? I think as it is specific to table and we know the actual dom strucre, kinda makes sense.
I guess my implementation in VS is not really nice, so you just want to go up the parent until you find scrollable element? additional cycle not something I look up for every row, but.. maybe?

Reply:
Previously @bentleyvk had the same check in case the user had the table in a scrollable container.

Should we somehow check this only once per table and set a flag that every row can access?

Reply:
I moved ref state to table